beacon-chain/blockchain/forkchoice/service.go

```diff
@@ -235,5 +237,5 @@ func (s *Store) Head(ctx context.Context) ([]byte, error) {
 
 // FinalizedCheckpt returns the latest finalized check point from fork choice store.
 func (s *Store) FinalizedCheckpt() *ethpb.Checkpoint {
-	return s.finalizedCheckpt
+	return proto.Clone(s.finalizedCheckpt).(*ethpb.Checkpoint)
```

Comment:
Why?

Reply:
We had a major bug where we used to do this in the blockchain service and these values would get mutated if we simply returned the pointer. We should be cloning this to be safe.